backend/src/agent_pipeline/agent.py

```diff
@@ -0,0 +1,56 @@
+from typing import List, Dict, Callable
+from .tool import ToolDefinition, ToolArgument, AgentGoal
+from .context import Context
+
+class Agent:
+    def __init__(self, agent_goal: AgentGoal, tool_definitions: List[ToolDefinition], tool_directory: Dict[str, Callable], context: Context=Context()):
+        """
+        Initialize the Agent with its goal, tool definitions, and executable tools.
+
+        Args:
+            agent_goal: The goal configuration for the agent.
+            tool_definitions: List of definitions for the tools available to the agent.
+            tool_directory: Dictionary mapping tool names to their callable functions.
+        """
+        # initializing the agent
+        self.agent_goal = agent_goal
```

Comment:
Looking at what's contained in `agent_goal`, maybe `details` is a better name? The corresponding class would be named `AgentDetails`. What do you think?